### ISSUE TITLE
Task fails on "Error: Operation not permitted (os error 1)" when run in parallel with other tasks

### ISSUE THREAD

**Issue by giltayar** [nrwl/nx#29137]
### Current Behavior

A task that runs in parallel with other tasks fails on "Error: Operation not permitted (os error 1)"

### Expected Behavior

The tasks succeeds, even if run in parallel with other tasks.

### Steps to Reproduce

1. Run ` nx build:dev @kusto/tridentkustoextension --verbose`
2. This task depends on `build`, which runs many tasks to build the monorepo packages, but also depends on `lint`, which lints the `tridentkustoextension` package.
3. If I run it with `--parallel 1` everything works, or if it is the only task to be left to run.
4. But if it needs to run in parallel with other tasks, it throws an exception (see "Failure Logs" for error).

Note that the weird formatting of the error is what I get and not a result of a bad copy/paste.

Also note that I'm not sure why this task fails and not others that also run in parallel.

Note that the `lint` task looks like this in the `package.json`: `"yarn run -T eslint --max-warnings 0 ./src ./test/integration`.

### Nx Report

```shell
Node           : 20.18.1
OS             : linux-x64
Native Target  : x86_64-linux
yarn           : 4.3.1

nx (global)  : 20.1.3
nx           : 20.1.4
@nx/devkit   : 20.1.2
typescript   : 5.6.2

---------------------------------------
Nx Powerpack
Licensed to **** for 20 users in 1 workspace until ********

@nx/powerpack-azure-cache  : 1.0.14

---------------------------------------
Registered Plugins:
@nx/powerpack-azure-cache
---------------------------------------
The following packages should match the installed version of nx
  - @nx/devkit@20.1.2

To fix this, run `nx migrate nx@20.1.4`





(note that running `nx migration nx@20.1.4` gives me this error: "The current directory isn't part of an Nx workspace.")
```

### Failure Logs

```shell
> nx run @kusto/tridentkustoextension:lint

                                          Error: Operation not permitted (os error 1)
                                                                                         at PseudoTerminal.fork (/home/giltayar/code/Azure-Kusto-WebUX/node_modules/nx/src/tasks-runner/pseudo-terminal.js:42:73)
                                                                                           at ForkedProcessTaskRunner.forkProcessWithPseudoTerminal (/home/giltayar/code/Azure-Kusto-WebUX/node_modules/nx/src/tasks-runner/forked-process-task-runner.js:140:45)
                 at ForkedProcessTaskRunner.forkProcess (/home/giltayar/code/Azure-Kusto-WebUX/node_modules/nx/src/tasks-runner/forked-process-task-runner.js:126:25)
                                               at TaskOrchestrator.runTaskInForkedProcess (/home/giltayar/code/Azure-Kusto-WebUX/node_modules/nx/src/tasks-runner/task-orchestrator.js:304:54)
                                                                        at TaskOrchestrator.applyFromCacheOrRunTask (/home/giltayar/code/Azure-Kusto-WebUX/node_modules/nx/src/tasks-runner/task-orchestrator.js:287:61)
                                                                                                  at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
                                                         at async TaskOrchestrator.executeNextBatchOfTasksUsingTaskSchedule (/home/giltayar/code/Azure-Kusto-WebUX/node_modules/nx/src/tasks-runner/task-orchestrator.js:85:13)
                                                                                                         at async Promise.all (index 1)
                 at async TaskOrchestrator.run (/home/giltayar/code/Azure-Kusto-WebUX/node_modules/nx/src/tasks-runner/task-orchestrator.js:61:9)
                           at async defaultTasksRunner (/home/giltayar/code/Azure-Kusto-WebUX/node_modules/nx/src/tasks-runner/default-tasks-runner.js:51:16) {
                                       code: 'GenericFailure'
                                                             }
```

### Package Manager Version

yarn 4.3.1

### Operating System

- [ ] macOS
- [x] Linux
- [x] Windows
- [ ] Other (Please specify)

### Additional Information

This is running in WSL under Windows.

**Comment by JakeDern** [comment]
@Cammisuli @FrozenPandaz I have the exact same issue, is there anyone investigating this? It's just for one project even when there are identical targets for other projects - I have the exact same errors that this issue is describing with the weird formatting, and running with `--parallel 1` fixes it completely.

**Comment by joshhunt** [comment]
👋 I also started running into this when adding an extra task. It reproduces 100%, but only after a `nx reset`. Calling the task again will make it succeed! Haven't got a _minimal_ repro, but I'm running into it in the mentioned Grafana PR.

**Comment by tlaak** [comment]
We have been hitting this bug as well in a monorepo with 60 packages, but it's not consistent. Might happen 2-3 times in a row, then works fine for days. Cache reset may or may not help, and the `--verbose` switch doesn't produce anything useful (no stacktrace gets printed out).

**Comment by MrF0o** [comment]
Having the same issue on wsl ubuntu, `--parallel 1` fixes it, but what's the point of using nx if we disable parallel builds.

**Comment by omasakun** [comment]
I ran into the same problem. The environment is WSL2.
The bug occurred when I tried to run two `vite build` at the same time in parallel.

I went back to previous versions to see if I could reproduce the bug, and it seems to have occurred [between `18.1.0-beta.9` and `18.1.0-beta.10`](https://github.com/nrwl/nx/compare/18.1.0-beta.9...18.1.0-beta.10) (possibly #22294)

- - - - -

In my environment, I can reproduce the bug with the minimal following configuration:

`package.json`
```json
{
  "name": "repro",
  "scripts": {
    "a": "echo",
    "b": "echo",
    "start": "npm i && nx reset && nx run repro"
  },
  "nx": {
    "targets": {
      "repro": {
        "command": "echo",
        "dependsOn": ["a", "b"]
      }
    }
  },
  "dependencies": {
    "nx": "18.1.0-beta.10"
  }
}
```

`nx.json`
```json
{}
```

Run `npm start` to reproduce the bug:

```
 NX   Running target repro for project repro and 2 tasks it depends on:

————————————————————————————————————————————————————————————————————————

> nx run repro:a

> nx run repro:b

> a
> echo

—————————————————————————————————————————————————————————————————————————

 NX   Running target repro for project repro and 2 tasks it depends on failed

Failed tasks:

- repro:b
```

**Comment by leosvelperez** [comment]
Hey all!

As some have shared, we traced and confirmed this issue to be a PTY-sharing regression introduced in 18.1.0-beta.10. In v21.0.0, we changed the forked task runner to use separate PTY instances instead of a shared singleton, which removes that problematic path.

Please update to v21+ (ideally to the latest version) to resolve the issue. If it still occurs, please create a new issue and share a fresh minimal repro and nx report so we can investigate any remaining environment-specific cases.
